source/dubregistry/dbcontroller.d

```diff
@@ -84,19 +84,24 @@ class DbController {
 
 		// drop old text index versions
 		db.runCommand(["dropIndexes": "packages", "index": "packages_full_text_search_index"]);
+		db.runCommand(["dropIndexes": "packages", "index": "packages_full_text_search_index_v2"]);
 
 		// add current text index
-		Bson[string] fts;
-		fts["key"] = ["$**": Bson("text")];
-		fts["name"] = "packages_full_text_search_index_v2";
-		fts["weights"] = [
-			"name": Bson(4),
-			"categories": Bson(3),
-			"versions.info.description" : Bson(3),
-			"versions.info.homepage" : Bson(1),
-			"versions.info.author" : Bson(1),
-			"versions.readme" : Bson(2),
+		immutable keyWeights = [
+			"name": 8,
+			"categories": 4,
```

Comment:
should categories really have a text index? After all it just stores the category ids which just happen to be human-readable right? I think it would decrease search quality for generic terms, the category should be a separate UI dropdown menu or something to filter by before searching.

Reply:
The idea is that if you enter "database" all packages in this category will be shown too (even if their name doesn't match).

Reply:
what about categories like std_aspirant, data_structures, optimized_mem, etc.

my point is that searching for the identifier of the category just happens to make sense for some existing categories, but will not work with other category labels.

It would be much better if we had a separate category search, or embedded it in the input field like `category:database` with autocompletion for it

Reply:
I think the plan was to get rid of these categories and replace them with tags anyhow as categories have a rather bad UX.